docs/warnings.md

```diff
@@ -51,3 +51,76 @@ However, some functions return RAW clauses into which may or may not accept untr
 If these are strings containing malicious code, a SQL injection attack can occur.
 Such functions, like the `where`, `sqlJoin`, or `sqlExpr` function, should escape the input. See [this page](/where/) for example.
 
+## Custom resolvers
+
+<div class="admonition danger">
+  <p class="first admonition-title">Warning</p>
+  <p class="last">
+    When writing custom resolvers on non-trivial fields, consider getting the value through the GraphQL default resolver.
+  </p>
+</div>
+
+GraphQL allows querying the same field through multiple aliases, which may use different arguments:
+
+```graphql
+{
+  users {
+    following { fullName }
+    andrews: following(name: "andrew") { fullName }
+  }
+}
+```
+
+Normally, `join-monster` would allow accessing the value as `source.following` in a custom resolver:
+
+```javascript
+const User = new GraphQLObjectType({
+  //...
+  fields: () => ({
+    //...
+    following: {
+      //...
+      resolve: (source, args, context, info) => {
+        return processUsers(source.following)
+      }
+    }
+  })
+})
+```
+
+However, if there are multiple conflicting aliases accessing the same relation, the `source` object will look something like this:
+
+```javascript
+{
+  following$: [
+    { fullName: 'andrew carlson' },
+    { fullName: 'matt elder' }
+  ],
+  following$matts: [
+    { fullName: 'matt elder' }
+  ],
+  following: (args, context, info) => { /* ... */ }
+}
+```
+
+The `source.following` property is now a function that will return the correct value. GraphQL's default resolver will detect the function and do the right thing.
+
+Therefore, we recommend getting the raw value through GraphQL's default resolver first when using a custom resolver on a non-trivial field:
```

Comment:
- "non-trivial field" ... this should have a more precise definition at least as best as you understand it.
- "recommend" ? ... perhaps too softly worded. What happens if you don't use the default resolver ... and (again) why are we calling the default resolver _explicitly_ when it was previously called _implicitly_?

Reply:
True, it might throw an error or do weird things if you don't use the default resolver and a query uses multiple aliases, because `source[fieldName]` would contain a function instead of the field value.

I changed the wording to "need to", and added a section explaining "non-trivial" values in more detail.